tests/test_routing.py

```diff
@@ -150,7 +152,8 @@ def test_detect_outlets(self):
             (0, 0), (1, 0), (2, 0), (3, 0),
             (3, 1),
             (0, 2),
-            (1, 3), (2, 3)}
+            (1, 3), (2, 3),
+            (511, 511)}
```

Comment:
It would be kinda nice to test cases on the edges of blocks because there's potential for weird stuff to happen there. I tried to do that in delineateit by mocking `iterblocks`: https://github.com/natcap/invest/blob/00b6c6e77f5f335e96d326d0c91d20dfcdc132f7/tests/test_delineateit.py#L411-L456
Up to you whether it's worth it to do that here.

Reply:
I agree, I was thinking of that and I couldn't think of an easy way to do it and decided it was probably fine. `mock`ing `iterblocks` is a great way to force that. I copied most of your test case you linked and finagled it to work with `detect_outlets` in 017339a9c59aae9e2884685b8466d141d8b35161.